### ISSUE TITLE
Why does "skiko.directx.gpu.priority" on "auto" prefer the "integrated"?

### ISSUE THREAD

**Issue by StefanOltmann** [JetBrains/compose-multiplatform#2897]
I read in https://blog.jetbrains.com/kotlin/2021/06/compose-for-desktop-milestone-4-released/#rendering that `skiko.directx.gpu.priority` with default setting `auto` will always try to use the integrated card by default.

Why is that? The dedicated graphic cards in most notebooks will always have better performance. So this does not make much sense to me.

If I set `skiko.directx.gpu.priority` to `discrete`, will this fall back to the integrated card if no discrete card is present? Or will this fail?

**Comment by eymar** [comment]
I tried setting `skiko.gpu.priority` when running on mac os. When no value provided then `auto` is a used as a default.

`integrated` - DEVICE::: skikoVersion=0.7.54, os=MacOS, api=METAL, deviceName=Intel(R) UHD Graphics 630
`discrete` - DEVICE::: skikoVersion=0.7.54, os=MacOS, api=METAL, deviceName=AMD Radeon Pro 5500M
`auto` - DEVICE::: skikoVersion=0.7.54, os=MacOS, api=METAL, deviceName=AMD Radeon Pro 5500M

So with `auto`, it choose the discrete card. And it makes sense to me.

@igordmn Could you please confirm that the behaviour described above is what you would expect? I see you've change the code related to this question some time ago - https://github.com/JetBrains/skiko/commit/bd79451ac87dbd4107d7bc460a831b6369805a6f

**Comment by eymar** [comment]
As for:

> Additional question:
> Is there a way to request the name of the currently used graphics device?
> I want to show that in my about screen.

I just copy paste my reply from the Slack:

> Window can be provided with create: () -> ComposeWindow factory - [https://github.com/JetBrains/compose-multiplatform-core/blob/814c6e48acb6a5054b161[…]desktopMain/kotlin/androidx/compose/ui/window/Window.desktop.kt](https://github.com/JetBrains/compose-multiplatform-core/blob/814c6e48acb6a5054b16156ca568ac97bad7b91f/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/Window.desktop.kt#L366)
> ComposeWindow can take [SkiaLayerAnalytics](https://github.com/JetBrains/skiko/blob/bd79451ac87dbd4107d7bc460a831b6369805a6f/skiko/src/commonMain/kotlin/org/jetbrains/skiko/SkiaLayerAnalytics.kt#L7). Where some info can be collected

**Comment by mikehearn** [comment]
That is normal for UI toolkits. 2D graphics should not require the performance of the discrete GPU and it drains a lot more power when it's activated. Discrete GPU is there more for games or proper 3D, not UI rendering.

**Comment by StefanOltmann** [comment]
Thank you for letting me know.

**Comment by StefanOltmann** [comment]
@igordmn Is what eymar wrote a bug or intentional? It's different to what's described in the blog post.

**Comment by orangy** [comment]
I agree with @mikehearn here, switching to discrete GPU shouldn't be needed for most desktop apps, and thus default (`auto`) should be using integrated whenever possible to conserve battery and avoid expensive GPU switching (affects app start up significantly, up to 400 ms on my machine). If an app is doing intensive work on the GPU, the developer should configure it explicitly to use discrete GPU.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2897">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
